shell/components/xen/xen.js

```diff
@@ -10,6 +10,26 @@ const html = (strings, ...values) => {
 
 Template.html = (...args) => Template.createTemplate(html.apply(null, args)); // eslint-disable-line prefer-spread
 
+const Debug = (Base, log) => class extends Base {
+  _wouldChangeProp(name, value) {
+    let result = true;
+    if (typeof value === 'object') {
+      result = true;
+    }
+    if (((name in this._pendingProps) && (this._pendingProps[name] !== value)) || (this._props[name] !== value)) {
+      result = true;
+      log('props', {[name]: value});
+    }
+    return result;
```

Comment:
isn't this always true?

Reply:
Yes, thank you, the default value should be false.

Reply:
re: Debug() and performance

Debug() isn't default, per se, you have to actively mix it in to a class to enable the extra logging. I haven't measured, but the cost is marginal since we were already doing most of this logging, it just wasn't standardized.

As stability improves, we can remove Debug mixin from various classes, or disable logging globally with a flag.

Reply:
@seefeldb passes the test!